--- drivers.old/gpu/drm/drm_gem.c	2011-08-19 15:34:42.000000000 +0200
+++ drivers/gpu/drm/drm_gem.c	2011-08-19 15:36:28.000000000 +0200
@@ -138,7 +138,7 @@
 	BUG_ON((size & (PAGE_SIZE - 1)) != 0);
 
 	obj->dev = dev;
-	obj->filp = shmem_file_setup("drm mm object", size, VM_NORESERVE);
+	obj->filp = shmem_file_setup("drm mm object", size, VM_NORESERVE, 0);
 	if (IS_ERR(obj->filp))
 		return -ENOMEM;
 
--- drivers.old/gpu/drm/ttm/ttm_tt.c	2011-08-19 15:34:42.000000000 +0200
+++ drivers/gpu/drm/ttm/ttm_tt.c	2011-08-19 15:36:09.000000000 +0200
@@ -544,7 +544,7 @@
 	if (!persistent_swap_storage) {
 		swap_storage = shmem_file_setup("ttm swap",
 						ttm->num_pages << PAGE_SHIFT,
-						0);
+						0, 0);
 		if (unlikely(IS_ERR(swap_storage))) {
 			printk(KERN_ERR "Failed allocating swap storage.\n");
 			return PTR_ERR(swap_storage);
