diff -U 3 -H -b -w -B -E -d -t -r -N -- var/tmp/portage/x11-base/nouveau-drm-20110801/work/master/drivers/gpu/drm/drm_gem.c tmp/drivers/gpu/drm/drm_gem.c
--- var/tmp/portage/x11-base/nouveau-drm-20110801/work/master/drivers/gpu/drm/drm_gem.c	2011-08-18 19:00:40.000000000 +0200
+++ tmp/drivers/gpu/drm/drm_gem.c	2011-08-18 19:04:55.000000000 +0200
@@ -138,7 +138,7 @@
         BUG_ON((size & (PAGE_SIZE - 1)) != 0);
 
         obj->dev = dev;
-        obj->filp = shmem_file_setup("drm mm object", size, VM_NORESERVE, 0);
+        obj->filp = shmem_file_setup("drm mm object", size, VM_NORESERVE);
         if (IS_ERR(obj->filp))
                 return -ENOMEM;
 
diff -U 3 -H -b -w -B -E -d -t -r -N -- var/tmp/portage/x11-base/nouveau-drm-20110801/work/master/drivers/gpu/drm/ttm/ttm_tt.c tmp/drivers/gpu/drm/ttm/ttm_tt.c
--- var/tmp/portage/x11-base/nouveau-drm-20110801/work/master/drivers/gpu/drm/ttm/ttm_tt.c	2011-08-18 19:00:18.000000000 +0200
+++ tmp/drivers/gpu/drm/ttm/ttm_tt.c	2011-08-18 19:04:55.000000000 +0200
@@ -544,7 +544,7 @@
         if (!persistent_swap_storage) {
                 swap_storage = shmem_file_setup("ttm swap",
                                                 ttm->num_pages << PAGE_SHIFT,
-                                                0, 0);
+                                                0);
                 if (unlikely(IS_ERR(swap_storage))) {
                         printk(KERN_ERR "Failed allocating swap storage.\n");
                         return PTR_ERR(swap_storage);
